test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_service_test.clj

```diff
@@ -744,6 +748,26 @@
              (is (= 200 (:status @response)))
              (is (= hello-body (:body @response))))))))))
 
+(deftest double-stop-test
+  (testing "if the stop lifecycle is called more than once, we handle that gracefully and quietly"
+    (let [log-messages (atom [])]
+      (tk-log-testutils/with-logging-to-atom tk-log/root-logger-name log-messages
+        (let [app (tk-bootstrap/bootstrap-services-with-config
+                   [jetty9-service]
+                   {:webserver {:port 8080}})]
+          (tk-app/stop app)
+          (tk-app/stop app)))
+      ;; if we try to unregister the mbeans more than once Jetty will log a bunch
+      ;; of very loud and terrifying InstanceNotFoundExceptions
+      (let [log-event-filter #(and (= :warn (:level %))
+                                   (not (nil? (:exception %)))
+                                   (re-find #"InstanceNotFoundException"
+                                            (.getClassName (:exception %))))
```

Comment:
Yeah, it turns out that Jetty is just swallowing the exceptions coming from the JDK MBean API and logging a warning, so there's nothing more concrete to test than log messages.

I think that what you are proposing would be more likely to yield false negatives, whereas mine is more likely to yield false positives.  False positives are probably worse, so I can look into changing it to what you're suggesting.